Automattic/simplenote-macos
# Interlinks Autocomplete UI: Mark II

### Details:
- Reimplements **SPWindow** in Swift: Adds support for the **ParentWindowDelegate** protocol
- Implements **AutocompleteWindow**

This new **NSWindow** subclass receives events from its parent **Window** entity, and allows us to redirect specific events (such as Arrows / Esc / Return) over to the Autocomplete helper window.

That way, the main Editor keeps the focus / firstResponder, and the Autocomplete UI reacts when needed.

@aerych sir!!! Can I trouble you with a **REALLY FUN** PR?
Thanks a looot!!

🔥 

Ref. #663

### Test
1. Open any note and enable Markdown
2. Type `[keyword` with `keyword` being a word contained within any random note

- [x] Verify the **Autocomplete Window** now shows the title of notes containing `keyword` in their contents
- [x] Verify that **Right / Left click** in the Parent Window (Editor) dismisses the Autocomplete Window
- [x] Verify that pressing **ESC** dismisses the Autocomplete Window
- [x] Verify that pressing **Arrow Up / Arrow Down** let you change the highlighted Autocomplete Row
- [x] Verify switching to **Light / Dark / System** themes cause the main app to instantaneously refresh
- [x] Verify switching to **Light / Dark / System** themes cause the Autocomplete Window to instantaneously refresh

### Release
These changes do not require release notes.
